src/Migrations/M044MigrateArticlesBlockToPostsListBlock.php

```diff
@@ -0,0 +1,367 @@
+    private static function check_is_valid_block(array $block): bool
+    {
+        // Check if the block is valid.
+        if (!is_array($block)) {
+            return false;
+        }
+
+        // Check if the block has a 'blockName' key.
+        if (!isset($block['blockName'])) {
+            return false;
+        }
+
+        // Check if the block is an Articles block.
+        return $block['blockName'] === Utils\Constants::BLOCK_ARTICLES;
+    }
+
+    /**
+     * Create a new Query block based on attributes of an Articles block.
+     *
+     * @param array $block - The current articles block.
+     * @return array - The new block.
+     */
+    private static function transform_block(array $block): array
+    {
+        $existing_block_attrs = self::get_posts_list_block_attrs($block);
+
+        $tags = $existing_block_attrs['tags'];
+        $posts_override = $existing_block_attrs['posts'];
+        $post_types = $existing_block_attrs['post_types'];
+        $layout_type = 'default';
+        $classname = 'list';
+        $rows = $existing_block_attrs['rows'];
+        $current_post_id = $existing_block_attrs['current_post_id'];
+        // add "Additional CSS class(es)" to new migrated block, if exists.
+        $additional_class = $existing_block_attrs['additional_class'];
+        if ($additional_class) {
+            $classname = $classname . ' ' . $additional_class;
+        }
+
+        $attrs = self::set_query_block_attrs($tags, $posts_override, $post_types, $layout_type, $rows, $current_post_id);
+
+        $inner_blocks = [
+            self::get_head_group_block($existing_block_attrs['title']),
+            self::get_paragraph_block($existing_block_attrs['description']),
+            self::get_query_no_results_block(),
+            self::get_post_template(),
+            self::get_buttons_block(),
+            self::get_nav_links_block(),
+        ];
+
+        return Utils\Functions::create_block_query(
+            $inner_blocks,
+            $attrs,
+            $classname
+        );
+    }
+
+    /**
+     * Get the attributes of a Post lists block.
+     *
+     * @param array $existing_block - The current Articles block.
+     * @return array - The attributes.
+     */
+    private static function get_posts_list_block_attrs(array $existing_block): array
+    {
+        $attrs = $existing_block['attrs'];
+
+        return [
+            'title' => isset($attrs['article_heading']) && !empty($attrs['article_heading']) ?
+                $attrs['article_heading'] : __('Related Articles', 'planet4-blocks'),
+            'description' => isset($attrs['articles_description']) ? $attrs['articles_description'] : '',
+            'tags' => isset($attrs['tags']) ? $attrs['tags'] : [],
+            'posts' => isset($attrs['posts']) ? $attrs['posts'] : [],
+            'post_types' => isset($attrs['post_types']) ? $attrs['post_types'] : [],
+            'rows' => isset($attrs['article_count']) ? $attrs['article_count'] : 3,
```

Comment:
Sorry this is me being picky, but calling this "rows" is kinda confusing since this is the `article_count` for the _Articles block_ and `per_page` for Posts List and is the number of posts to display.